dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/supplier/UserSupplier.java

```diff
@@ -55,16 +58,34 @@
     @NonNull
     private final IdentifiableObjectManager manager;
 
+    @NonNull
+    private final UserService userService;
+
     @Override
     public void preheatAdd( TrackerImportParams params, TrackerPreheat preheat )
     {
         Set<String> userUids = params.getEvents().stream()
             .filter( Objects::nonNull )
             .map( Event::getAssignedUser )
+            .filter( Objects::nonNull )
+            .map( UserInfo::getUid )
             .filter( CodeGenerator::isValidUid )
             .collect( Collectors.toSet() );
 
-        preheat.put( TrackerIdentifier.UID,
+        Set<String> usernames = params.getEvents().stream()
+            .filter( Objects::nonNull )
+            .map( Event::getAssignedUser )
+            .filter( Objects::nonNull )
+            .map( UserInfo::getUsername )
+            .collect( Collectors.toSet() );
```

Comment:
should we filter out StringUtils.isBlank ones? like you do above with `CodeGenerator::isValidUid`?

Reply:
Done